### ISSUE TITLE
Can't create ElastiCache with VPC security group

### ISSUE THREAD

**Issue by tiyberius** [ansible/ansible-modules-core#1088]
##### Issue Type:

Bug Report
##### Ansible Version:

$ ansible --version
ansible 1.9.0.1
##### Environment:

Mac OS 10.10.2
##### Summary:

I'd like to create a VPC, a security group, and an ElastiCache cluster. The ElastiCache cluster should be Redis, and I want it to be in a VPC security group.
##### Steps To Reproduce:

Run the following playbook

```

---
- name: Demonstrate Ansible ElastiCache bug
  hosts: localhost
  connection: local
  gather_facts: False
  tasks:

  - name: Create a VPC
    local_action:
      module: ec2_vpc
      state: present
      cidr_block: 10.0.0.0/16
      resource_tags:
        Name: Test VPC please delete
      subnets:
      - cidr: 10.0.1.0/24
        az: us-east-1a
      - cidr: 10.0.2.0/24
        az: us-east-1c
      route_tables:
        - subnets:
          - 10.0.1.0/24
          routes:
            - dest: 0.0.0.0/0
              gw: igw
      internet_gateway: True
      region: us-east-1
      wait: true
    register: our_vpc
    until: vpc
    retries: 2
    delay: 5

  - name: Create the security group for Redis in ElastiCache
    local_action:
      module: ec2_group
      name: Redis Security Group
      description: Created for demonstration of elasticache module bug
      vpc_id: "{{ our_vpc.vpc_id }}"
      region: us-east-1
      rules: []
    ignore_errors: yes
    register: redis_security_group

  - name: Create a Redis cluster in ElastiCache
    elasticache:
      name: redis-cluster
      state: present
      engine: redis
      cache_engine_version: 2.8.6
      node_type: cache.r3.large
      num_nodes: 1
      security_group_ids: "{{ redis_security_group.group_id }}"
      region: us-east-1
      zone: us-east-1a
```
##### Expected Results:

I expected a Redis ElastiCache cluster to be created.
##### Actual Results:

When I run the aforementioned playbook, I get

```
ansible-playbook elasticache_bug.yml                                                                                                                                   [15:18:41]

PLAY [Demonstrate Ansible ElastiCache bug] ************************************

TASK: [Create a VPC] **********************************************************
changed: [localhost -> 127.0.0.1]

TASK: [Create the security group for Redis in ElastiCache] ********************
ok: [localhost -> 127.0.0.1]

TASK: [Create a Redis cluster in ElastiCache] *********************************
failed: [localhost] => {"failed": true}
msg: {"Error":{"Code":"InvalidParameterValue","Message":"Use of cache security groups is not permitted in this API version for your account.","Type":"Sender"},"RequestId":"bc6cb385-dbc8-11e4-b886-6fbf8e293bce"}

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/twork/elasticache_bug.retry

localhost                  : ok=2    changed=1    unreachable=0    failed=1
```

and no ElastiCache cluster :(

**Comment by timmahoney** [comment]
You can probably use this module that I made to get the job done:

https://github.com/ansible/ansible-modules-core/pull/1137

**Comment by tiyberius** [comment]
Hey @timmahoney thanks for creating the module! I realized just now that I know how to create bug reports, but don't actually know how to test the fix. I'll try to figure that out today or tomorrow and re-run the playbook.

**Comment by bcoca** [comment]
closing via merged #1137
